File: Python/Binary_Search.py
def binary_search(arr, target):
    left, right = 0, len(arr) - 1
    
    while left <= right:
        mid = left + (right - left) // 2 
        
        if arr[mid] == target:
            return mid
        elif arr[mid] < target:
            left = mid + 1
        else:
            right = mid - 1
    
    return -1 

arr = list(map(int, input("Enter a sorted list of numbers separated by spaces: ").split()))
target = int(input("Enter the target number: "))

result = binary_search(arr, target)
print(f"Element found at index: {result}" if result != -1 else "Element not found")

File: JavaScript/binary_search.js
function binarySearch(arr, key) {
  let start = 0;
  let end = arr.length - 1;

  while (start <= end) {
    let mid = Math.floor(start + (end - start) / 2);

    if (arr[mid] === key) {
      return mid;
    } else if (arr[mid] > key) {
      end = mid - 1;
    } else {
      start = mid + 1;
    }
  }

  return -1;
}

const arr = [2, 3, 4, 10, 40];
const key = 10;

const result = binarySearch(arr, key);
if (result !== -1) {
  console.log("Element found at index:", result);
} else {
  console.log("Element not found");
}
